### PR TITLE
Filter notification candidates by 24-hour presence freshness

### PR DESCRIPTION
**Pull request by justinrooks**
# Summary
Apply the shared 24-hour hard-stale presence cutoff to notification candidate queries and expose matching eligibility metrics in the operator dashboard. The branch also adds regression coverage for H3/UGC filtering and dashboard aggregation, plus compatibility defaults for stored snapshots.

# What Changed
- Exclude presence records older than 24 hours from H3 and UGC notification candidate queries.
- Promote the shared hard-stale threshold for reuse by delivery and dashboard code.
- Add candidate-query eligibility and hard-stale presence counts/rates to stored and rendered operator dashboard metrics.
- Preserve decoding compatibility for older dashboard snapshots by backfilling new metric fields.
- Add regression tests for cutoff-boundary candidate filtering, dashboard aggregation, and snapshot/rendered response behavior.

# User Impact
Users with presence data older than 24 hours are no longer selected as notification candidates, reducing delivery to stale locations. Operators gain dashboard visibility into candidate-query eligibility and hard-stale exclusions.

# Testing
- `swift test` built successfully and executed 427 tests across 56 suites, but reported 1 failing issue.
- A follow-up focused test command could not run because the sandbox blocked Swift manifest/module-cache access.

# Notes
- The working tree contains an unrelated uncommitted change in `docs/Sql/Device.sql`; it is excluded from this PR.
- This PR is opened as a draft for review because the full-suite failure remains to be isolated.

### CODE REVIEW

File: Sources/App/Jobs/NotificationSendJob.swift [hunk 1]
```diff
@@ -337,6 +342,7 @@ extension NotificationSendJob {
             WHERE i.is_active = TRUE
               AND i.is_subscribed = TRUE
               AND i.apns_device_token <> ''
+              AND p.captured_at >= \(bind: cutoff)
```
[the remark this answers is not in the document]
Comment (justinrooks) [reply Sources/App/Jobs/NotificationSendJob.swift:345]: Good catch on the accounting change—it is intentional for #62. Hard-stale presence is now excluded before candidate resolution specifically to avoid loading, decoding, logging, and attempting notification_missed_decisions inserts for that population. A match set containing only >24-hour presence therefore has zero eligible candidates and is recorded as zeroCandidates.
allCandidatesStaleLocation and stale-miss persistence remain available for candidates returned by the query but rejected by the application freshness policy, such as non-granted authorization or a boundary candidate aging past the cutoff before evaluation.
Preserving per-alert accounting for query-excluded rows would require retaining those rows or adding stale-match query work, both of which conflict with the optimization and the explicit non-goal in #62. The dashboard’s aggregate hardStalePresenceCount provides visibility into the excluded population instead.
